main.tf

```diff
@@ -1,3 +1,25 @@
+locals {
+  # Validate that only 'arm64' architecture is used with 'g' processor instances to ensure compatibility.
+  # https://docs.aws.amazon.com/ec2/latest/instancetypes/instance-type-names.html
+  is_instance_compatible = (
```

Comment:
`g` can be present in the instance family: `G – Graphics intensive`. This is obviously a weird choice for an SSM agent, but I think we still need to ensure we validate the processor family field.
According to the docs, this is a third position, and we can use the function `element(list, index)` to check this.
Thoughts on making this check stricter?

Reply:
The `g` isn't representing the instance family, it's representing the processor family of that instance, and it represents `AWS Graviton processors` which is AWS' ARM based processors

Reply:
Oh I see what you mean now, that the validation would be valid because it currently only checks for any 'g's. But having the stricter check will ensure that the processor family flag has a 'g'.

Yeah that makes sense then, definitely will make this check stricter.

Reply:
Just added those stricter validation, and also tinkered with some native TF tests to cover those edge cases. 

That's a great catch about the instance family, thanks! Yeah it would be strange to use for an SSM agent, just looking at their lowest pricing. But always nice to have these validations in place. 

![image](https://github.com/user-attachments/assets/d54e752a-ef30-44c6-b54b-2b62948fdb8a)

Reply:
Looks great! 🤩